gwpy/timeseries/timeseries.py

```diff
@@ -287,6 +287,11 @@ def psd(self, fftlength=None, overlap=None, window='hann',
         # get method
         method_func = spectral.get_method(method)
 
+        # work-around to propagate 'window' argument intact
+        # for LPSD averaging method
+        if "method" in kwargs and kwargs["method"] == "lpsd":
+            kwargs["window_"] = window
```

Comment:
The reason is that when window argument "comes back" to the method when it's called from fft_registry, it is already a converted to numpy array instead of being a string. LPSD package wants a pointer to window function instead of numpy array. I can create it from string just before calling LPSD but not from numpy array. Therefore I added an extra kwarg to keep that string intact through the calls and then use it. Happy to implement it nicer if there's a way